test/datetime.cc

```diff
@@ -583,9 +583,56 @@ TEST(DateTime, TestSecondOfWeek) {
   EXPECT_EQ(a, e) << "Wrong second of week";
 }
 
+TEST(DateTime, DiffCaching) {
+  // no cache
```

Comment:
first we check that the changes work even with out cache but checking the time difference between NY and LA to be -3 hours